### ISSUE TITLE
Bug: no-useless-constructor when readonly params

### ISSUE THREAD

**Issue by claudio-di-sciacca** [eslint/eslint#15962]
### Environment

Node version: 18.0.0
npm version: 8.6.0
Local ESLint version: 8.14.0
Global ESLint version: 8.14.0
Operating System: Debian GNU/Linux 11 (bullseye)

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```json
{
  "compilerOptions": {
    "target": "es2022",
    "module": "commonjs",
    "outDir": "./dist",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true
  },
  "include": [
    "lib/**/*",
    "samples/**/*",
    "test/**/*"
  ]
}
```
</details>

```ts
class Test {
  constructor (
      readonly url: string,
      readonly verbose?: boolean
  ) {}
}

// eslint-disable-next-line no-new
new Test('url', false)
```

### What did you expect to happen?

No error to be shown since declaring "readonly" values inside a class constructor is not useless.

### What actually happened?

Useless constructor. eslint([no-useless-constructor](https://eslint.org/docs/rules/no-useless-constructor))

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

If it's not urgent I might be able to work on the fix.

**Comment by mdjermanovic** [comment]
Hi @claudio-di-sciacca, thanks for the issue!

This rule doesn't support TypeScript. You should use [@typescript-eslint/no-useless-constructor](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-useless-constructor.md) rule instead.

```json
{  
    "no-useless-constructor": "off",
    "@typescript-eslint/no-useless-constructor": "error"
}
```
